test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -66,8 +66,8 @@ class LoginProtectionTest < ActionController::TestCase
       session[:shopify_domain] = "foobar"
       sess = stub(url: 'https://foobar.myshopify.com')
       ShopifyApp::SessionRepository.expects(:retrieve).returns(sess).once
-      get :second_login, params: { shop: 'other_shop' }
```

Comment:
Separately I think there's a bug with `ShopifyApp::Utils.sanitize_shop_domain`
It doesn't like `_` characters - is that expected?

Reply:
`_` is not a legal character in domain names, so yes this is expected